src/audio/buffer.c

```diff
@@ -313,7 +313,7 @@ void buffer_attach(struct comp_buffer *buffer, struct list_item *head, int dir)
 	if (further_buffers_exist)
 		dcache_invalidate_region(needs_sync, sizeof(struct list_item));
 	/* no dirty cache lines exist for this buffer yet, no need to write back */
-	dcache_invalidate_region(uncache_to_cache(list), sizeof(*list));
+	dcache_writeback_invalidate_region(uncache_to_cache(list), sizeof(*list));
```

Comment:
Hmm, this is now in conflict with the comment on L315 . So either we need to change the comment or there is something else going on.

@lyakh I believe you added the comment?